python/sg_qwidgets/sg_qwidgets.py

```diff
@@ -297,11 +297,100 @@ class SGQProgressBar(QtGui.QProgressBar):
 
 
 class SGQPushButton(QtGui.QPushButton):
-    """
-    ShotGrid wrapper class for QSGQPushButtonwidget.
+    """ShotGrid wrapper class for QSGQPushButtonwidget."""
 
-    No additional functionality for this wrapper class, other than declaring it as a subclass.
-    """
+    def __init__(self, *args, **kwargs):
+        """Initialize the button with the default style for push buttons."""
+
+        super().__init__(*args, **kwargs)
+
+        # Set default style for SGQPushButton.
+        self.setStyleSheet("")
+
+        # Set cursor to pointing hand when hovering over any SGQPushButton
+        self.setCursor(QtGui.QCursor(QtCore.Qt.PointingHandCursor))
+
+    def enterEvent(self, event):
+        """Override method to add hover effect when mouse enters the button."""
+
+        hover_effect = self.__get_hover_effect()
+        self.setGraphicsEffect(hover_effect)
+
+    def leaveEvent(self, event):
+        """Override method to remove hover effect when mouse leaves the button."""
+
+        self.setGraphicsEffect(None)
+
+    def __get_hover_effect(self):
+        """Get the hover effect for the button."""
+
+        # Create a glow effect for the button
+        hover_effect = QtGui.QGraphicsDropShadowEffect(self)
+        hover_effect.setBlurRadius(3)
+        hover_effect.setOffset(0, 0)
+        hover_effect.setColor(QtGui.QColor(204, 204, 204, 250))
+        return hover_effect
+
+
+class SGSubmitPushButton(SGQPushButton):
+    """Custom class for 'Submit' buttons styled for SG."""
+
+    def __init__(self, *args, **kwargs):
+        """Initialize the button with the default style for 'Submit'."""
+
+        super().__init__(*args, **kwargs)
+
+        qss = self.styleSheet()
+        qss += """
+            SGSubmitPushButton {
+                color: rgb(255, 255, 255);
+                background-color: rgb(24, 167, 227);
+                border: none;
+                border-radius: 4px;
+                padding: 4px 12px 4px 12px;
+            }
+        """
```

Comment:
Personal question: why sometimes the styling of a button is done by qss code but the `__get_hover_effect` defined previously used class-based styling?

Reply:
From my experience, it is not consistent across Toolkit, whether style is applied using qss stylesheets or programmatically.

In this case, I prefer using qss stylesheets because it is much easier/quicker to apply style to a button for simple attributes (e.g. color, border, etc.), otherwise we have to override and re-implement the button `paintEvent` method, which is more complicated and more work to do. Though I prefer qss stylesheets, there are some more sophisticated styles that cannot be achieved using qss - this particular button hover effect I could not find a way to achieve it with qss.